src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaCacheClientFetchFactory.java

```diff
@@ -243,6 +244,8 @@ public MessageConsumer newStream(
             }
 
             final KafkaFilterCondition condition = cursorFactory.asCondition(filters);
+            final KafkaAge maximumAge = filters.anyMatch(a -> a.conditions().anyMatch(c -> c.age().get() != KafkaAge.HISTORICAL)) ?
```

Comment:
Do we need to check `c.kind() == AGE` before calling `c.age()` in the matcher?
Also we can probably hoist this stateless lambda into a member variable of the factory.